tests/image/dataframe_iterator_test.py

```diff
@@ -0,0 +1,653 @@
+import os
+import random
+import shutil
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from PIL import Image
+
+from keras_preprocessing.image import dataframe_iterator
+from keras_preprocessing.image import image_data_generator
+
+
+@pytest.fixture(scope='module')
+def all_test_images():
+    img_w = img_h = 20
+    rgb_images = []
+    rgba_images = []
+    gray_images = []
+    for n in range(8):
+        bias = np.random.rand(img_w, img_h, 1) * 64
+        variance = np.random.rand(img_w, img_h, 1) * (255 - 64)
+        imarray = np.random.rand(img_w, img_h, 3) * variance + bias
+        im = Image.fromarray(imarray.astype('uint8')).convert('RGB')
+        rgb_images.append(im)
+
+        imarray = np.random.rand(img_w, img_h, 4) * variance + bias
+        im = Image.fromarray(imarray.astype('uint8')).convert('RGBA')
+        rgba_images.append(im)
+
+        imarray = np.random.rand(img_w, img_h, 1) * variance + bias
+        im = Image.fromarray(
+            imarray.astype('uint8').squeeze()).convert('L')
+        gray_images.append(im)
+
+    return [rgb_images, rgba_images, gray_images]
+
+
+def test_dataframe_iterator(all_test_images, tmpdir):
+    num_classes = 2
+
+    # save the images in the tmpdir
+    count = 0
+    filenames = []
+    filepaths = []
+    filenames_without = []
+    for test_images in all_test_images:
+        for im in test_images:
+            filename = "image-{}.png".format(count)
+            filename_without = "image-{}".format(count)
+            filenames.append(filename)
+            filepaths.append(os.path.join(str(tmpdir), filename))
+            filenames_without.append(filename_without)
+            im.save(str(tmpdir / filename))
+            count += 1
+
+    df = pd.DataFrame({
+        "filename": filenames,
+        "class": [str(random.randint(0, 1)) for _ in filenames],
+        "filepaths": filepaths
+    })
+
+    # create iterator
+    iterator = dataframe_iterator.DataFrameIterator(df, str(tmpdir))
+    batch = next(iterator)
+    assert len(batch) == 2
+    assert isinstance(batch[0], np.ndarray)
+    assert isinstance(batch[1], np.ndarray)
+    generator = image_data_generator.ImageDataGenerator()
+    df_iterator = generator.flow_from_dataframe(df, x_col='filepaths')
+    df_iterator_dir = generator.flow_from_dataframe(df, str(tmpdir))
+    df_sparse_iterator = generator.flow_from_dataframe(df, str(tmpdir),
+                                                       class_mode="sparse")
+    if np.isnan(df_sparse_iterator.classes).any():
```

Comment:
When does this can happen?

Reply:
I don't know. I just moved a test that was already there. But I converted into an assertion